designs/virtualization/4.x/create-vm-dropdowns/create-vm-dropdowns.md

```diff
@@ -0,0 +1,112 @@
+---
+parent: Virtualization
+version: 4.x
+---
+
+# create VM dropdowns
+
+This item will cover various updates to the VM creation wizard
+- Adding descriptions to the dropdown fields (using this pattern https://github.com/patternfly/patternfly-design-kit/issues/218)
+- Renaming fields
+- Reordering fields
+
+
+![Create VM - General-default](img/1-0.jpg)
+Landing on the General step, Default mode.
+Section titles added, and fields reorgenized and renamed.
+'General' section:
+- 'Name'
+- 'Description'
+'Resources & configuration' section'
+- 'Operating system'
+- 'Configuration' (previously Workload profile)
+- 'Flavor (CPU and Memory)' (previously only 'flavor')
+'Operating system image source' section
+- 'Boot operating system via' (previously 'source')
+
+## General step
+
+![Create VM - General](img/1-1.jpg)
+Upon selecting an operating system, the configuration and Flavor (CPU and Memory) fields will be enabled and their defaults will be selected.
```

Comment:
@suomiy 
The design displayed here hadn't changed anything really from the current implementation:
- User sees the OS field first
- User can start by selecting any of the fields to filter the others (for example, Flavor)

In any case, the user needs to select all the fields to result in one common template.
Since this is a UI, some fields must come before others.

The flow you mention is a "broken" one since the users are frustrated that none of the OS options fit their needs. No matter what they select, it is a compromise made by an educated guess at best. If this is a common case where users have to improvise by selecting what OS they suspect will fit best, we need to address that, either by adding "Other OS" in the OS field (I know it's a problem), or changing the entire UI completely.
I can try to provide a solution if we do think this is a significant use case, but for now, I'm not trying to change anything in the current implementation, just reorder some fields (switch between flavor and WorkloadProfile/Configuration and renaming others.
If we agree we need to change this behavior, by UI or Back-end, then it should be covered in a different PR

Reply:
+1 I think you are making a good point @suomiy but I'm reluctant to say we should make any assumptions that are based in user feedback. Does it makes sense to you to hold on this until we can some better research done?

Reply:
> Does it makes sense to you to hold on this until we can some better research done?

> User can start by selecting any of the fields to filter the others (for example, Flavor)

If it can stay enabled like this, then it's okay. Unless we know adding these limitations would be  useful, we should be careful adding them there.

Reply:
> In any case, the user needs to select all the fields to result in one common template.

Generally yes, but in case of custom flavor there might be for example 4 possible templates and we just pick one. This might have some unforseen consequences for the user

Reply:
I'm switching the Workload Type and Flavor fields to be always enabled.

>Generally yes, but in case of custom flavor there might be for example 4 possible templates and we just pick one. This might have some unforseen consequences for the user

So this generally means we have to have a Common Template field, as a safety net for back-end potential problems. I'll see how we can surface that in the UI

Reply:
Adding Common template